libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt

```diff
@@ -0,0 +1,157 @@
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Edge
+
+        if (id != other.id) return false
+        if (level != other.level) return false
+        if (probability != other.probability) return false
+        if (heading != other.heading) return false
+        if (length != other.length) return false
+        if (out != other.out) return false
+        if (parent != other.parent) return false
+        if (frc != other.frc) return false
+        if (speed != other.speed) return false
+        if (ramp != other.ramp) return false
+        if (motorway != other.motorway) return false
+        if (bridge != other.bridge) return false
+        if (tunnel != other.tunnel) return false
+        if (toll != other.toll) return false
+        if (names != other.names) return false
+        if (curvature != other.curvature) return false
+        if (geometry != other.geometry) return false
+        if (speedLimit != other.speedLimit) return false
+        if (laneCount != other.laneCount) return false
+        if (meanElevation != other.meanElevation) return false
+        if (countryCode != other.countryCode) return false
+        if (stateCode != other.stateCode) return false
+
+        return true
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun hashCode(): Int {
+        var result = id.hashCode()
+        result = 31 * result + level.hashCode()
+        result = 31 * result + probability.hashCode()
+        result = 31 * result + heading.hashCode()
+        result = 31 * result + length.hashCode()
+        result = 31 * result + out.hashCode()
+        result = 31 * result + parent.hashCode()
+        result = 31 * result + frc.hashCode()
+        result = 31 * result + speed.hashCode()
+        result = 31 * result + ramp.hashCode()
+        result = 31 * result + motorway.hashCode()
+        result = 31 * result + bridge.hashCode()
+        result = 31 * result + tunnel.hashCode()
+        result = 31 * result + toll.hashCode()
+        result = 31 * result + names.hashCode()
+        result = 31 * result + curvature.hashCode()
+        result = 31 * result + geometry.hashCode()
+        result = 31 * result + speedLimit.hashCode()
+        result = 31 * result + laneCount.hashCode()
+        result = 31 * result + meanElevation.hashCode()
+        result = 31 * result + countryCode.hashCode()
+        result = 31 * result + stateCode.hashCode()
+        return result
+    }
+
+    /**
+     * Returns a string representation of the object.
+     */
+    override fun toString(): String {
+        return "Edge(" +
+            "id=$id, " +
+            "level=$level, " +
+            "probability=$probability, " +
+            "heading=$heading, " +
+            "length=$length, " +
+            "out=$out, " +
```

Comment:
we would have the same issue here wouldn't we, `out` could lead to a loop edge right?

Reply:
Not sure, didn't run into any. Also that's not new / something we've added as part of this PR so we should've run into it already? 🤔